test/integration/testdata/gen-rerunfailed-missing.golden

```diff
@@ -0,0 +1,44 @@
+Error: invalid argument "testdata/results-missing-e2e.tar.gz" for "--rerun-failed" flag: failed to find file "plugins/e2e/sonobuoy_results.yaml" in archive
```

Comment:
I always wonder about the golden files.  They seem like a like a lot of work to maintain.  I guess they make validation easy.

Reply:
They are _soooooooooooooooooooo_ easy to maintain in comparison to other types of tests.

Its very easy to make new test cases: copy/paste the test case struct and just change the details (name of file, command to test, etc). Then run `update_local` and check that it looks good.

Then no matter what UX or logic changes happen, if you want to make a small or big change you just have to run the `update_local` command again and verify the output looks good.

Otherwise you're stuck manually trying to verify random parts of large output and miss lots of little things too. For instance, this PR was failing the goldenfile tests at first because the help menu apparently was listing the `image-pull-policy` values but not sorting them so about 1/3 of the time they'd swap order.

They are definitely verbose, but since you dont write/maintain the contents _manually_ they are so nice.

Reply:
If you werent aware, I dont ever run commands and copy/paste into those files. The tests are made to handle an `--update` flag. If that it set then rather than testing the output against the goldenfile, it saves its output into the goldenfile.

So you can write 40 test cases of output you expect to be OK, run the tests with the update flag (use the update_local script function), and it creates all those files for you. After a quick look, you have documented the exact output of 40 test cases.

For `gen` commands especially this is really nice since the muli-yaml document would be painful to have to try and parse for certain details (there may be some tests that still do that and I think (1) they miss important data that could be tested and (2) I think they comment about how awkward it is)

Reply:
@johnSchnake that is great, thanks for the explanation. 
I never made it into that part of the code, maybe at some point we can do a walk through so I can understand it a bit.